lib/elixir/lib/enum.ex

```diff
@@ -899,14 +899,18 @@ defmodule Enum do
   end
 
   def find_index(enumerable, fun) when is_function(fun, 1) do
-    res =
-      Enumerable.reduce(enumerable, {:cont, 0}, fn(entry, acc) ->
-        if fun.(entry), do: {:halt, acc}, else: {:cont, acc + 1}
-      end)
+    res = Enumerable.reduce(enumerable, {:cont, {0, 0}}, fn(entry, acc) ->
```

Comment:
I think the start value should be `{:cont, {:not_found, 0}}`,
otherwise it will fail below with a `CaseClauseError` for empty enumerables.

Maybe we could also add a test with an empty enumerable.